web/src/app/components/login-page/login-page.component.ts

```diff
@@ -1,31 +1,60 @@
-import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { AuthService } from '../../services/auth.service';
+import { UserService } from '../../services/user.service';
 
 @Component({
   selector: 'app-login-page',
   templateUrl: './login-page.component.html',
-  styleUrls: ['./login-page.component.scss'],
-  encapsulation: ViewEncapsulation.None,
+  styleUrls: ['./login-page.component.scss']
 })
-
 export class LoginPageComponent implements OnInit {
 
-  constructor(private authService: AuthService) { }
+  constructor(
+    private authService: AuthService,
+    private userService: UserService,
+  ) {}
 
-  ngOnInit() {
-  }
+  public hasError = false;
+  public errorMessage = '';
+
+  ngOnInit() {}
 
   loginWithGithub() {
     this.authService.signInWithGithub()
-      .subscribe((result) => {
-        console.log('result', result); // TODO: remove after firestore implementation
+      .subscribe((user) => {
+        if (!user) {
+          this.hasError = true;
+          this.errorMessage = 'Something went wrong. Please try again';
+          return;
+        }
+        if (user) {
```

Comment:
I dont think this condition is needed, because the first condition has a `return`.

This would also work...?

```typescript
if (!user) {
   this.hasError = true;
   this.errorMessage = 'Something went wrong. Please try again';
   return;
 }

this.saveUser(user, 'github');
```

However, should this happen earlier? Should the check happen in `formatUserResponse` before `normaliseGithubUser`?

Reply:
@eddiejaoude Thanks for pointing it out. I have added a check for null object before the format response is done. Even then we need the above condition to show the error message in the login screen.